### PR TITLE
feat: DR overview

### PR DESCRIPTION
**Pull request by killme2008**
## What's Changed in this PR

Adds an introduction to DR solution in GreptimeDB. part of #996 .
Preview link https://feature-da-intro.greptime-docs.pages.dev/nightly/user-guide/operations/disaster-recovery/overview
## Checklist

- [ ] Please ensure that the content in `summary.yml` matches the current document structure when you changed the document structure.
- [x] This change requires follow-up update in localized docs.

## Summary by CodeRabbit

- **Documentation**
  - Added detailed explanations on disaster recovery concepts, including RTO, RPO, and GreptimeDB architecture components.
  - Introduced various disaster recovery solutions: Standalone, Dual Active-Standby, Cross-Region Single-Cluster, and Backup & Restore.
  - Provided comparisons of disaster recovery solutions based on error tolerance, RPO, RTO, TCO, and scenarios.
  - Added new documentation files for specific DR solutions in both English and Chinese for enhanced understanding.

### CODE REVIEW

**Review by zhaoyingnan01 — COMMENTED** [review]
有的形状是实边框 而有的是虚边框这个可以做的处理吗

**Comment by killme2008** [comment]
> 有的形状是实边框 而有的是虚边框这个可以做的处理吗

有统一的规范吗

**Comment by nicecui** [comment]
> docs/public/RTO-RPO-explain.png

Why does time have two directions?

**Review by nicecui — COMMENTED** [review]
```
- DR solution for Standalone
- DR solution based on Dual Active-Standby
- DR solution based on cross-region deployment in a single cluster 
- DR solution based on BR 
```

These aren't the same level of hierarchy. Logically, standalone corresponds to cluster.
Also, the `Dual Active-Standby solution` is a solution for standalone. So the section titles would confuse users.

Maybe the following hierarchy would be better (I removed the `DR solution` for concise) :
```
- Standalone
  - Remote WAL and object storage
  - Dual Active-Standby
- Cluster
  - Cross-region replication
  - Backup & restore
```

File: docs/nightly/en/user-guide/operations/disaster-recovery/overview.md [hunk 1]
```diff
@@ -1,3 +1,113 @@
 # Overview
 
-* [Backup & restore data](./back-up-&-restore-data.md)
\ No newline at end of file
+GreptimeDB is a distributed database designed to withstand disasters. It provides different solutions for disaster recovery (DR).
+
+## Basic Concepts
+
+* **Recovery Time Objective (RTO)**: refers to the maximum acceptable amount of time that a business process can be down after a disaster occurs before it negatively impacts the organization.
+* **Recovery Point Objective (RPO)**: refers to the maximum acceptable amount of data loss measured in time before the disaster occurs. It indicates the time interval during which data might be lost due to a major incident.
```
[the remark this answers is not in the document]
Comment (killme2008) [reply docs/nightly/en/user-guide/operations/disaster-recovery/overview.md:8]: I think it's very clear: `the maximum acceptable amount of data loss measured in time`.

File: docs/nightly/en/user-guide/operations/disaster-recovery/overview.md [hunk 2]
```diff
@@ -1,3 +1,113 @@
...
+* **WAL**: persists the unflushed Memtable data in memory. It will be truncated when the Memtable is flushed into SSTable files. It can be local disk-based(local WAL) or Kafka cluster-based(remote WAL).
+* **Object Storage**: persists the SSTable data and index.
+
+The GreptimeDB stores data in object storage such as [AWS S3](https://docs.aws.amazon.com/AmazonS3/latest/userguide/DataDurability.html) or its compatible services, which is designed to provide 99.999999999% durability and 99.99% availability of objects over a given year. And services such as S3 provide [replications in Single-Region or Cross-Region](https://docs.aws.amazon.com/AmazonS3/latest/userguide/replication.html), which is naturally capable of DR.
+
+At the same time, the WAL component is pluggable, e.g. using Kafka as the WAL service that offers a mature [DR solution](https://www.confluent.io/blog/disaster-recovery-multi-datacenter-apache-kafka-deployments/).
+
+### BR
```
[the remark this answers is not in the document]
Comment (killme2008) [reply docs/nightly/en/user-guide/operations/disaster-recovery/overview.md:40]: I should figure out when I first time use it.

File: docs/nightly/en/user-guide/operations/disaster-recovery/overview.md [hunk 3]
```diff
@@ -1,3 +1,113 @@
...
+The DR solution based on BR provides an RPO depending on the backup frequency and an RTO that varies with the size of the data to be restored.
+
+Read [Backup & restore data](./back-up-&-restore-data.md) for details, and we plan to provide a BR tool in-house for this solution.
+
+### Solution Comparison
+
+By comparing these DR solutions, users can decide on the final option based on their specific scenarios, requirements, and investment.
+
+
+|     DR solution | Error Tolerance Objective |  RPO | RTO | TCO | Scenarios | Notes |
```
[the remark this answers is not in the document]
Comment (killme2008) [reply docs/nightly/en/user-guide/operations/disaster-recovery/overview.md:97]: `Total Cost of Ownership`, commonly referred to in the software industry.

File: docs/nightly/en/user-guide/operations/disaster-recovery/overview.md [hunk 4]
```diff
@@ -1,3 +1,113 @@
...
+| ------------- | ------------------------- | ----- | ----- | ----- | ---------- | --------|
+|  DR solution for Standalone| Single-Region | Backup Interval | Minute or Hour level | Low | Low requirements for availability and reliability in small scenarios |  Recommended to use remote WAL and object storage to achieve high availability and reliability |
+|  DR solution based on Dual Active-Standby| Cross-Region | 0 | Minute level | Medium | High requirements for availability and reliability in small-to-medium scenarios |  Remote WAL and Object storage are optional, commercial feature|
```
[the remark this answers is not in the document]
Comment (killme2008) [reply docs/nightly/en/user-guide/operations/disaster-recovery/overview.md:100]: It's not a standalone but a special cluster version.

**Comment by killme2008** [comment]
> ```
> - DR solution for Standalone
> - DR solution based on Dual Active-Standby
> - DR solution based on cross-region deployment in a single cluster 
> - DR solution based on BR 
> ```
> 
> These aren't the same level of hierarchy. Logically, standalone corresponds to cluster. Also, the `Dual Active-Standby solution` is a solution for standalone. So the section titles would confuse users.
> 
> Maybe the following hierarchy would be better (I removed the `DR solution` for concise) :
> 
> ```
> - Standalone
>   - Remote WAL and object storage
>   - Dual Active-Standby
> - Cluster
>   - Cross-region replication
>   - Backup & restore
> ```

No, the Dual Active-Standby solution is not a solution for standalone. What standalone means is always the single process of GreptimeDB, but the Dual Active-Standby solution is not.

The BR is suitable for both standalone and clusters, not just only for cluster.

File: docs/nightly/en/user-guide/operations/disaster-recovery/overview.md [hunk 5]
```diff
@@ -1,3 +1,113 @@
...
+## Component architecture
+
+### GreptimeDB
+
+Before digging into the specific DR solution, let's explain the architecture of GreptimeDB components in the perspective of DR:
+![Component-architecture](/Component-architecture.png)
+
+GreptimeDB is designed with a cloud-native architecture based on storage-compute separation：
+* **Frontend**:  the ingestion and query service layer, which forwards requests to Datanode and processes, and merges responses from Datanode.
+* **Datanode**:  the storage layer of GreptimeDB, and is an LSM storage engine. Region is the basic unit for storing and scheduling data in Datanode. A region is a table partition, a collection of data rows. The data in region is saved into Object Storage(such as AWS S3). Unflushed Memtable data is written into WAL and can be recovered in DA.
```
[the remark this answers is not in the document]
Comment (killme2008) [reply docs/nightly/en/user-guide/operations/disaster-recovery/overview.md]: Typo.

**Comment by killme2008** [comment]
> > docs/public/RTO-RPO-explain.png
> 
> Why does the time have two directions? It is hard to understand the difference between RTO and RPO. It seems they are time points in the image, but maybe not?

The directory is wrong, already fixed it.  There are tons of articles that explain RTO and RPO in detail, you can google it.

**Review by nicecui — APPROVED** [review]

**Review by nicecui — APPROVED** [review]

File: docs/nightly/en/user-guide/operations/disaster-recovery/overview.md [hunk 6]
```diff
@@ -1,3 +1,126 @@
...
+
+![Dual-active-standby](/Dual-active-standby.png)
+
+In some edge or small-to-medium scale scenarios, or if you lack the resources to deploy remote WAL or object storage, Dual Active-Standby offers a better solution compared to Standalone DR. By replicating requests synchronously between two standalone nodes, high availability is ensured. The failure of any single node will not lead to data loss or a decrease in service availability even when using local disk-based WAL and data storage.
+
+Deploying nodes in different regions can also meet region-level DR requirements, but the scalability is limited.
+
+:::tip NOTE
+
+**the Dual Active-Standby is a commercial solution.**
```
Comment (tisonkun) [inline docs/nightly/en/user-guide/operations/disaster-recovery/overview.md:78]:
```suggestion
**Dual Active-Standby is only available in GreptimeDB Enterprise.**
```

... and considering link to the Enterprise page.

File: docs/nightly/en/user-guide/operations/disaster-recovery/overview.md [hunk 7]
```diff
@@ -1,3 +1,126 @@
...
+### DR solution for GreptimeDB Standalone
+
+If the Standalone is running on the local disk for WAL and data, then:
+* RPO: depends on backup frequency.
+* RTO: doesn't make sense in standalone mode, mostly depends on the size of the data to be restored, your failure response time, and the operational infrastructure.
+
+But if running the Standalone with remote WAL and object storage, there is a better DR solution:
+![DR-Standalone](/DR-Standalone.png)
+
+Write the WAL to the Kafka cluster and store the data in object storage, so the database itself is stateless. In the event of a disaster affecting the standalone database, you can restore it using the remote WAL and object storage. This solution allows for an RPO of zero, and an RTO measured in minutes.
```
Comment (tisonkun) [inline docs/nightly/en/user-guide/operations/disaster-recovery/overview.md:64]:
```suggestion
Write the WAL to the Kafka cluster and store the data in object storage, so the database itself is stateless. In the event of a disaster affecting the standalone database, you can restore it using the remote WAL and object storage. This solution can achieve **RPO=0** and **RTO in minutes**.
```

File: docs/nightly/en/user-guide/operations/disaster-recovery/overview.md [hunk 8]
```diff
@@ -1,3 +1,126 @@
...
+* **WAL**: persists the unflushed Memtable data in memory. It will be truncated when the Memtable is flushed into SSTable files. It can be local disk-based (local WAL) or Kafka cluster-based (remote WAL).
+* **Object Storage**: persists the SSTable data and index.
+
+The GreptimeDB stores data in object storage such as [AWS S3](https://docs.aws.amazon.com/AmazonS3/latest/userguide/DataDurability.html) or its compatible services, which is designed to provide 99.999999999% durability and 99.99% availability of objects over a given year. And services such as S3 provide [replications in Single-Region or Cross-Region](https://docs.aws.amazon.com/AmazonS3/latest/userguide/replication.html), which is naturally capable of DR.
+
+At the same time, the WAL component is pluggable, e.g. using Kafka as the WAL service that offers a mature [DR solution](https://www.confluent.io/blog/disaster-recovery-multi-datacenter-apache-kafka-deployments/).
+
+### Backup and restore
```
Comment (tisonkun) [inline docs/nightly/en/user-guide/operations/disaster-recovery/overview.md:46]: Shall we talk about how to do backup and restore?

File: docs/nightly/zh/user-guide/operations/disaster-recovery/overview.md [hunk 9]
```diff
@@ -1,3 +1,143 @@
...
+![Dual-active-standby](/Dual-active-standby.png)
+
+在某些边缘或中小型场景中，或者如果你没有资源部署远程 WAL 或对象存储，双活-备份相对于 Standalone 的灾难恢复提供了更好的解决方案。
+通过在两个独立节点之间同步复制请求，确保了高可用性。
+即使使用基于本地磁盘的 WAL 和数据存储，任何单个节点的故障也不会导致数据丢失或服务可用性降低。
+
+在不同区域部署节点也可以满足区域级灾难恢复要求，但可扩展性有限。
+
+:::tip 注意
+**双活-备份是商业解决方案。**
```
Comment (tisonkun) [inline docs/nightly/zh/user-guide/operations/disaster-recovery/overview.md:90]:
```suggestion
**双活-备份功能仅在 GreptimeDB 企业版中提供。**
```

File: docs/nightly/zh/user-guide/operations/disaster-recovery/overview.md [hunk 10]
```diff
@@ -1,3 +1,143 @@
...
+* RPO：取决于备份频率。
+* RTO：在 Standalone 没有意义，主要取决于要恢复的数据大小、故障响应时间和操作基础设施。
+
+但是如果使用远程 WAL 和对象存储运行 Standalone，有一个更好的 DR 解决方案：
+
+![DR-Standalone](/DR-Standalone.png)
+
+将 WAL 写入 Kafka 集群，并将数据存储在对象存储中，因此数据库本身是无状态的。
+在影响独立数据库的灾难事件发生时，你可以使用远程 WAL 和对象存储来恢复它。
+此解决方案允许零 RPO 和以分钟为单位的 RTO。
```
Comment (tisonkun) [inline docs/nightly/zh/user-guide/operations/disaster-recovery/overview.md:75]:
```suggestion
此方案能实现 RPO=0 和分钟级 RTO。
```

File: docs/nightly/en/user-guide/operations/disaster-recovery/overview.md (hunk 8 above)
Comment (tisonkun) [reply docs/nightly/en/user-guide/operations/disaster-recovery/overview.md:46]: Well. I saw it below at "DR solution based on BR". Perhaps providing an intra-doc link.

**Review by tisonkun — APPROVED** [review]
LGTM. Minor comments.

**Pull request merged**